File: sdk/java/dagger-codegen-maven-plugin/src/main/java/io/dagger/codegen/introspection/SchemaVisitor.java
package io.dagger.codegen.introspection;

import java.util.List;

public interface SchemaVisitor {

  void visitScalar(Type type);

  void visitObject(Type type);

  void visitInput(Type type);

  void visitEnum(Type type);

  void visitVersion(String version);

  void visitIDAbles(List<Type> types);
}

File: sdk/php/src/Codegen/Introspection/SchemaVisitor.php
<?php

namespace Dagger\Codegen\Introspection;

use GraphQL\Type\Definition\EnumType;
use GraphQL\Type\Definition\InputObjectType;
use GraphQL\Type\Definition\ObjectType;
use GraphQL\Type\Definition\ScalarType;

interface SchemaVisitor
{
    public function visitScalar(ScalarType $type): void;

    public function visitObject(ObjectType $type): void;

    public function visitInput(InputObjectType $type): void;

    public function visitEnum(EnumType $type): void;
}
